### PR TITLE
Feature/dark mode support

### PR DESCRIPTION
**Pull request by snypelife**
This pull request enables dark mode styles to resolve https://github.com/keithclark/gadebugger/issues/40.

Here's a preview:

image

Let me know what you think. Thanks!

### CODE REVIEW

**Comment by keithclark** [comment]
This is a fantastic start on supporting themes in Chrome — thank you!

There are a few visual things that need addressing before I can merge this - most are cosmetic changes:

* The grunt build task fails due to two linting errors
* In the default (light) theme, the selection colour is blue when devtools has focus and it turns grey when focus is lost. However in this dark theme update, the selection colour is always grey. Looking at the behaviour in other tabs, selections should be an orange colour when devtools has focus.
* The grey rule colour between panels and items seems brighter than in the other devtools panels.
* The beacon list (centre panel) item text is coloured (and bold?), rather than using the default text colour. The same colour also appears to be used in the left sidebar for the tracker domain.
* The toolbar icons and icon splitters also need looking at as they are also too bright.

As a rule, I try and base the layout/colours on the network and profile tabs as they have the closest UI layout to GA debugger.

If you could address those changes then I'd be more than happy merge this. If you're not able to make the updates, then I'd be more than happy to pick them up for you so we can get this merged and deployed in the next release.

Thanks again for your contribution, it's appreciated!

**Comment by snypelife** [comment]
No problem at all, I'll fix those styles up.

As for the lint error, that was actually occurring for me even on master. I could only get it to run by using `--force`.

image

**Comment by snypelife** [comment]
With tweaked styles:

image

image

**Comment by snypelife** [comment]
I can look at fixing that lint error too if you want? Figured to minimize the scope of this PR, I wouldn't do that at the same time. 

Up to you though ¯\_(ツ)_/¯

**Comment by snypelife** [comment]
lint error appears to have been because this https://github.com/jshint/jshint/issues/2922#issuecomment-219263558

**Comment by keithclark** [comment]
That's looking a lot better and thanks for the lint fix :)

Could you move the [theme class assignment](https://github.com/snypelife/gadebugger/blob/43bf28c89ea36a30104fce818661aced7dbabe5a/src/chrome/js/api.js#L8-L10) back to `window.ui.js`? I think it made more sense there. You can fix the linting error by adding `chrome` to the list of globals on the first line of that file.

What do you think about moving some of the overrides from `gadebugger.darkmode.css` into the related CSS files so they sit alongside their light theme decerlations? It would make maintenance easier in the future? For example:

* [These toolbar styles](https://github.com/snypelife/gadebugger/blob/43bf28c89ea36a30104fce818661aced7dbabe5a/src/chrome/css/gadebugger.darkmode.css#L17-L22) could go in ui.toolbar.css
* [These glyph styles](https://github.com/snypelife/gadebugger/blob/43bf28c89ea36a30104fce818661aced7dbabe5a/src/chrome/css/gadebugger.darkmode.css#L24-L26) could go in ui.glyph.css
* [This splitview style](https://github.com/snypelife/gadebugger/blob/43bf28c89ea36a30104fce818661aced7dbabe5a/src/chrome/css/gadebugger.darkmode.css#L55-L57) could go in ui.splitview.css
* and so on...?

**Comment by snypelife** [comment]
Sorry, I thought it made sense there too. The lint error threw me off so I moved it 😆 . I'll move it back.

As for breaking up the theme styles, I wondered about that and decided to go with the separate file approach. Reason being that if Chrome eventually opens up more theme options (purely speculative at this point), then it would be fairly simple to add support for more themes via a new theme css file with different colour assignments. 

From there you could also break out the default styles into their own and have positional/layout styles and theme styles. 

But your point is totally valid and making an architectural change like that on speculation is probably overkill.

Anyway, I'll drop a couple more commits to fix those things up 👍

File: src/chrome/css/gadebugger.css [hunk 1]
```diff
@@ -8,10 +8,19 @@ body > .splitview {
     flex: 1;
 }
 
+.dev-tools-theme-dark {
+    color: rgb(213, 213, 213);
```
Comment (keithclark) [inline src/chrome/css/gadebugger.css]: This isn't needed as it's overridden on the next line

**Comment by keithclark** [comment]
Thanks for those changes, it's much easier to manage the CSS changes now they're grouped together. I only have one more review comment.

Once that's fixed, I'll get your work merged into master. I have a Firefox bug to finish off and some Chrome UI tweaks to make prior to releasing the next version of GADebugger.

Regarding the architectural changes to CSS, I'm considering switching the CSS over to Sass and creating a grunt task to build the relevant theme variations for each browser. Both Firefox and Chrome support theming in devtools. This is a low priority task though.

File: src/chrome/css/gadebugger.css (hunk 1 above)
Comment (snypelife) [inline src/chrome/css/gadebugger.css]: doh! forgot i put that.

**Comment by snypelife** [comment]
Awesome, I'll look out for the update 👍 

If you want an assist at all with the bug or an initial spike at switching to Sass, I'd be down to help out.

**Comment by keithclark** [comment]
I've merged your updates into a [2.1.6 integration branch](https://github.com/keithclark/gadebugger/compare/2.1.6-dev?expand=1).

Once again, thanks for your contribution. I'll credit you accordingly.

**Review by muratkalenderoglu — COMMENTED** [review]
[](url)

**Pull request merged**